share/functions/help.fish

```diff
@@ -23,58 +22,58 @@ function help --description 'Show help for the fish shell'
 	# by the help function defined below.
 	#
 	set -l graphical_browsers htmlview x-www-browser firefox galeon mozilla konqueror epiphany opera netscape rekonq google-chrome chromium-browser
-	set -l text_browsers htmlview www-browser links elinks lynx w3m
 
-	if type -q "$BROWSER"
-		# User has manually set a preferred browser, so we respect that
-		set fish_browser $BROWSER
-
-		# If browser is known to be graphical, put into background
-		if contains -- $BROWSER $graphical_browsers
-			set fish_browser_bg 1
-		end
+	if type -q "$fish_help_browser[1]"
+		# User has set a fish-specific help browser. This overrides the
+		# browser that may be defined by $BROWSER. The $fish_help_browser
+		# variable may be an array containing a browser name plus options.
+		set fish_browser $fish_help_browser
 	else
-		# Check for a text-based browser.
-		for i in $text_browsers
-			if type -q -f $i
-				set fish_browser $i
-				break
-			end
-		end
-
-		# If we are in a graphical environment, check if there is a graphical
-		# browser to use instead.
-		if test "$DISPLAY" -a \( "$XAUTHORITY" = "$HOME/.Xauthority" -o "$XAUTHORITY" = "" \)
-			for i in $graphical_browsers
+		set -l text_browsers htmlview www-browser links elinks lynx w3m
+
+		if type -q "$BROWSER"
```

Comment:
I know this is what the prior version did but as I commented above I believe this should simply be `if set -q BROWSER`.